LoopKit/Alert.swift

```diff
@@ -94,15 +94,19 @@ public struct Alert: Equatable {
         case sound(name: String)
     }
     public let sound: Sound?
+
+    /// Any parameters used to dynamically construct foreground or background content encoded as a JSON object string
+    public let parameters: String?
```

Comment:
I think I'm still not convinced that the best way to store this here at this level is a JSON-encoded (i.e. post-serialized) string.  I think my preference still would be as a Dictionary, (i.e. `[String: Any]`).  Feel free to make a `typealias` if it helps.

Reply:
The problem here is that decoding the resultant JSON string stored in Core Data won't necessarily decode back into the expected data types. For example (and just as an example, not that you'd do this), if you stored an `Alert.Trigger` in the dictionary, then after it did a round-trip from `Alert` to Core Data (via `StoredAlert`) back to `Alert` what once was an `Alert.Trigger` is now a `String` (because `Alert.Trigger` encodes as a `String` in JSON). The only way I see to make this consistent and not error prone would be to create a brand new type of dictionary (like `UserDefaults) that **only** accepts the core JSON data types. That seems like way too much unnecessary complication for me.

Reply:
But that doesn't need to be enforced, does it?  Can't we just declare that, by convention, the Values in this dictionary have to be valid JSON (or plist, whichever you prefer) types?  This is not that uncommon, in my experience.

Reply:
That would not be my preference. If everyone else feels otherwise, I'll change it.

Reply:
It just seems wrong to me to put a serialized string in here as part of the LoopKit API.  I'm fine moving forward if I'm the only one who objects, but I still do object.

Reply:
Yea, I'm going to need more details here for both of your suggestions. I'll add it to the list of things to discuss at dev sync.